### PR TITLE
InsertAt (issue #306)

### PR DESCRIPTION
**Pull request by leandromoh**

### CODE REVIEW

**Comment by leandromoh** [comment]
any problem with this PR ?

File: MoreLinq.Test/InsertAtTest.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+        [TestCase(7)]
+        [TestCase(8)]
+        [TestCase(9)]
+        public void InsertAtWithIndexGreaterThanSourceLengthLazy(int count)
+        {
+            Enumerable.Range(0, count)
+                      .InsertAt(new[] { 97, 98, 99 }, count + 1)
+                      .Take(count)
+                      .ToList();
```
Comment (fsateler) [inline MoreLinq.Test/InsertAtTest.cs]: There is no expectation in this test

File: MoreLinq/InsertAt.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        /// <summary>
+        /// Inserts the elements of a sequence into another sequence, at the specified index.
+        /// </summary>
+        /// <typeparam name="T">Type of the elements of the source sequence.</typeparam>
+        /// <param name="source">The source sequence.</param>
+        /// <param name="second">The sequence that will be inserted into source.</param>
+        /// <param name="index">The zero-based index at which the new elements should be inserted.</param>
+        /// <returns>
+        /// A sequence that contains the elements of <paramref name="source"/>
+        /// Plus the elements of <paramref name="second"/>, at the specified index.
```
Comment (fsateler) [inline MoreLinq/InsertAt.cs]: `Plus` does not have to be capitalized here. Also, I'd add a 'inserted': `, inserted at the specified index.`

File: MoreLinq/InsertAt.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        /// </returns>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// Thrown eagerly if <paramref name="index"/> is negative, and lazily 
+        /// if it is greater than the length of <paramref name="source"/>
```
Comment (fsateler) [inline MoreLinq/InsertAt.cs]: This comment contradicts test `InsertAtWithIndexGreaterThanSourceLengthLazy`

File: MoreLinq/InsertAt.cs [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        /// </summary>
+        /// <typeparam name="T">Type of the elements of the source sequence.</typeparam>
+        /// <param name="source">The source sequence.</param>
+        /// <param name="second">The sequence that will be inserted into source.</param>
+        /// <param name="index">The zero-based index at which the new elements should be inserted.</param>
+        /// <returns>
+        /// A sequence that contains the elements of <paramref name="source"/>
+        /// Plus the elements of <paramref name="second"/>, at the specified index.
+        /// </returns>
+        /// <exception cref="ArgumentOutOfRangeException">
```
Comment (fsateler) [inline MoreLinq/InsertAt.cs]: There's the `ArgumentNullException` notes missing. Also, split this into two `<exception>` tags, one for each condition.

File: MoreLinq/InsertAt.cs [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        /// Thrown eagerly if <paramref name="index"/> is negative, and lazily 
+        /// if it is greater than the length of <paramref name="source"/> 
+        /// (after completely iterate the elements of the <paramref name="source"/>).
```
Comment (fsateler) [inline MoreLinq/InsertAt.cs]: s/iterate/iterating/ .

**Comment by leandromoh** [comment]
@fsateler all adjusts done! thanks!

File: MoreLinq.Test/InsertAtTest.cs [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq.Test
+{
+    using System.Linq;
```
Comment (atifaziz) [inline MoreLinq.Test/InsertAtTest.cs]: Remove this import. See #351 for more on why.

**Comment by leandromoh** [comment]
@atifaziz remove `System.Linq`. we are in time for 2.8.0?

**Comment by leandromoh** [comment]
@atifaziz renamed to `Insert`, added readme.md and on release notes. we think are ready for merge.

**Comment by atifaziz** [comment]
@leandromoh Thanks!

I know @fsateler is busy so I have gone ahead and merged without awaiting his approval. He had some documentation comments that seem to have been addressed.

**Comment by atifaziz** [comment]
:memo: This PR was for issue #306.

**Pull request merged**